File: src/test/java/testdata/java/VariableTestPOJO.java
package testdata.java;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

public class VariableTestPOJO {

    private List<SimpleTestPOJO<String>> listField = new ArrayList<>();

    public VariableTestPOJO(SimpleTestPOJO<Integer> cParameter) {

    }

    private void pojoMethod(SimpleTestPOJO<String> mParameter) {

        SimpleTestPOJO<Data> localVariable = new SimpleTestPOJO<>();
    }
}

class SimpleTestPOJO<T> {

    private int anInt = 0;

    private String string = "";

    private BigDecimal bigDecimal = BigDecimal.ZERO;

    private int[] ints = {0};

    private Set<Integer> linkedHashSet = new LinkedHashSet<>();

    private T data;

}

class Data {

    @com.fasterxml.jackson.annotation.JsonIgnore
    private String username;
    private String password;
}

File: src/test/java/testdata/kotlin/VariableTestPOJO.kt
package testdata.kotlin

import com.fasterxml.jackson.annotation.JsonIgnore
import java.math.BigDecimal

class VariableTestPOJO(cParameter: SimpleTestPOJO<Int?>?) {
    private val listField: List<SimpleTestPOJO<String>> = ArrayList()
    private fun pojoMethod(mParameter: SimpleTestPOJO<String>) {
        val localVariable = SimpleTestPOJO<Data>()
    }
}

class SimpleTestPOJO<T> {
    private val anInt = 0
    private val string = ""
    private val bigDecimal = BigDecimal.ZERO
    private val ints = intArrayOf(0)
    private val linkedHashSet: Set<Int> = LinkedHashSet<Int>()
    private val data: T? = null
}

class Data {
    @JsonIgnore
    private val username: String? = null
    private val password: String? = null
}